mpegts-segmenter/src/segmenter.rs

```
@@ -230,6 +237,16 @@ impl<S: SegmentStorage> Segmenter<S> {
                         }
                     }
                     segment.pts = pes_packet_header.and_then(|h| h.optional_header).and_then(|h| h.pts);
+                    if let Some((previous_segment, mut previous_segment_info)) = self.previous_segment.take().zip(self.previous_segment_info.take()) {
+                        if let Some((prev_pts, curr_pts)) = previous_segment.pts.zip(segment.pts) {
+                            let duration = (curr_pts + PTS_ROLLOVER_MOD - prev_pts) % PTS_ROLLOVER_MOD;
+                            // set the segment duration only if it's within a reasonable value.
+                            if duration < (self.config.min_segment_duration.as_secs_f64() * VIDEO_PTS_BASE as f64 * 3.0) as u64 {
```

We can raise an error at this point if the duration is an unreasonable value, but it results in the loss of two segments(current one and previous one). Additionally, the unit test `test_segmenter_restart` will fail due to this error. Alternatively, by setting the segment duration to `None`, the streaming can continue, and we can include an error message in the av-service log if it's necessary.

---

My feeling is that it's okay to discard both segments if the pts delta is crazy:

* We should make every effort on the agent side anyway to have correct, monotonically increasing timestamps by e.g. having a NTP-synced wall clock, taking care with our pts on Continuum input switch, etc., so that HA works as well as it can, so that sync between streams works well, etc.
* Not sure there's anything reasonable the player can do with a segment of unknown duration anyway. I really want to get rid of any temptation for the service, fusion feed, the player, etc. to add in heuristics and then have things just get off. The player folks need everything to be frame-accurate for AR, so I don't want to fudge it anywhere.
* Eventually, I'm tempted to switch from segments being stored as MPEG-TS to segments being stored as fMP4/CMAF. For those, we have to come up with a duration.

Strictly speaking, we don't *have* to drop the entire previous segment if the duration is crazy; we could basically go back to the previous video frame or something. But that probably requires an `SegmentStorage` interface change or something and doesn't seem work the complexity.

@epingel2 do you have opinions on this?

---

I don't know which one have more impact on the player, dropping segments or the duration of unknown segments. @maredudd what are your thoughts on this?